AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift

```diff
@@ -0,0 +1,432 @@
+
+        let postsCopy = posts
+        Task {
+            var postsSyncedToCloudCount = 0
+            let mutationEvents = Amplify.DataStore.observe(Post.self)
+            do {
+                for try await mutationEvent in mutationEvents {
+                    guard postsCopy.contains(where: { $0.id == mutationEvent.modelId }) else {
+                        return
+                    }
+
+                    if mutationEvent.mutationType == MutationEvent.MutationType.create.rawValue,
+                       mutationEvent.version == 1 {
+                        postsSyncedToCloudCount += 1
+                        await postsSyncedToCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
+            }
+        }
+
+        let capturedPosts = posts
+
+        DispatchQueue.concurrentPerform(iterations: concurrencyLimit) { index in
+            Task {
+                _ = try await Amplify.DataStore.save(capturedPosts[index])
+            }
+        }
+        await waitForExpectations([postsSyncedToCloud], timeout: networkTimeout)
+
+        let postsDeletedLocally = asyncExpectation(description: "All posts deleted locally",
+                                                   expectedFulfillmentCount: concurrencyLimit)
+
+        let postsDeletedFromCloud = asyncExpectation(description: "All posts deleted and synced to cloud",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        
+        Task {
+            var postsDeletedFromCloudCount = 0
+            let mutationEvents = Amplify.DataStore.observe(Post.self)
+            do {
+                for try await mutationEvent in mutationEvents {
+                    guard capturedPosts.contains(where: { $0.id == mutationEvent.modelId }) else {
+                        return
+                    }
+
+                    if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
+                              mutationEvent.version == 1 {
+                        await postsDeletedLocally.fulfill()
+                    } else if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
+                              mutationEvent.version == 2 {
+                        postsDeletedFromCloudCount += 1
+                        await postsDeletedFromCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
+            }
+        }
+        
+        DispatchQueue.concurrentPerform(iterations: concurrencyLimit) { index in
+            Task {
+                try await Amplify.DataStore.delete(capturedPosts[index])
+            }
+        }
+        
+        await waitForExpectations([postsDeletedLocally, postsDeletedFromCloud], timeout: networkTimeout)
+    }
+
+    // MARK: - Helpers
+
+    func validateSavePost() async throws {
```

Comment:
I didn't see this used, can it be removed or is this useful for future stress testing use cases?